setup.cfg

```diff
@@ -1,4 +1,87 @@
-# This file should be used as the pep8/pycodestyle and autopep8 config
+###############################################################################
+# Setuptools
+###############################################################################
+
+[metadata]
+name = DIRAC
+description = DIRAC is an interware, meaning a software framework for distributed computing.
+long_description = file: README.rst
+license = GPL-3.0-only
+classifiers =
+    License :: OSI Approved :: GNU General Public License v3 (GPLv3)
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+
+[options]
+python_requires = >=3.8
+package_dir=
+    =src
+packages = find:
+install_requires =
```

Comment:
It would be useful if someone can take a look at the dependencies and see if I've missed anything or got the client/server/testing split wrong.

Version constraints should also be added before the first non-pre release but that can wait for a little while.

Reply:
`environment-py3.yml`, `requirements.txt`, `docs/requirements.txt` and `docs/requirements_py3.txt` will disappear _relatively_ soon. We will then be left with only three places:

* `setup.cfg`: Runtime dependencies of DIRAC that are Python packages themselves, validated whenever DIRAC is installed
* `environment.yml`: Any packages that need to be available to run the tests (excluding integration for now, though that's not actually a problem now as everything is available from conda-forge), i.e. not limited to just Python packages so this can included singularity/python itself/openssl/...
* `constructor.yml` in DIRACOS2: Packages which will be included in the DIRACOS2 tarball
* `meta.yaml`: Allows a conda package to be built for DIRAC, not a big deal as it can be automatically generated from the PyPI metadata, that is itself generated from the `setup.cfg` file.

While there will be some duplication between the above they each serve distinct roles any any effort to de-duplicate them further will be more work than it's worth.

Reply:
`constructor.yml` will be a subset of `environment.yml`, right? And `setup.cfg` will also be a (different) subset of `environment.yml`.

Reply:
Yes though package names in `setup.cfg` will be the names on PyPI which might differ slightly from the name of the conda-package. For example `graphviz` is the Python bindings on PyPI but the CLI application on conda-forge (`python-graphviz` is the conda-forge name for the Python bindings).